feat(j-s): show case number on infocard

thorhildurt opened the pull request:
# [Asana task](https://app.asana.com/0/1199153462262248/1209154897751443)

## What
- Show merge case number which exist outside of the system

## Why
- To show the case number of the case being merged to 

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Enhanced merge case display logic in the InfoCard component.
  - Added new helper functions to improve merge case information retrieval.

- **Improvements**
  - Refined text handling for external merge cases.
  - Improved conditional rendering of merge case details.

- **Localization**
  - Added new localization string for external merge case display.

The changes improve the information presentation and handling of merge case details in the judicial system web application.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

oddsson commented:
You also need to change`InfoCardClosedIndictment` to show this if `workingCase.mergeCaseNumber` is set :)

thorhildurt commented:
I fixed the check and  updated the label which was a super tiny task in this [asana ticket](https://app.asana.com/0/1199153462262248/1209150155925146). 

Screenshot: 
![Screenshot 2025-01-16 at 16 18 11](https://github.com/user-attachments/assets/e45cce87-05bb-47d2-91dc-3ea3ce01c40d)

oddsson commented on apps/judicial-system/web/src/components/InfoCard/InfoCardClosedIndictment.tsx:
```diff
@@ -40,6 +41,9 @@ const InfoCardClosedIndictment: FC<Props> = (props) => {
   )?.created
 
+  const showMergeCaseItem =
+    isNonEmptyArray(mergeCase.values) && !!mergeCase.values[0]
```
There are now two ways to merge a case with another. 

1. Merge it with a case that is in RVG. In this case, the id of the case in RVG is set in the `mergeCaseId` field in DB. 
2. Merge it with a case that is not in RVG. In this case, the court case number is set in the `mergeCaseNumber` field in db.

To check if we should show the merge case id we'd need to check if either of these fields are set, i.e.

`
const showMergeCaseItem = workingCase.mergeCase || workingCase.mergeCaseNumber
`

thorhildurt replied:
As discussed, we ensure to check this in the hook where `mergeCase` `Item` is configured to avoid double checking if a case has internal or external case merge. But I will move this check to the hook since I think this is simply a utility to decide weather to render an item or not which is useful for other info cards 👍

oddsson reviewed (changes requested):
👀
